rakelib/utils.rake

```diff
@@ -0,0 +1,58 @@
+# run a command, pipe output through 'xcpretty' and store the output in CI artifacts
+def xcpretty(cmd, task, subtask = '')
+  name = (task.name + (subtask.empty? ? '' : "_#{subtask}")).gsub(/[:-]/, "_")
+  command = [*cmd].join(' && ')
+  xcpretty = `which xcpretty`
+
+  if ENV['CI']
+    sh "set -o pipefail && (#{command}) | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\" | xcpretty --color --report junit --output \"#{ENV['CIRCLE_TEST_REPORTS']}/xcode/#{name}.xml\""
+  elsif xcpretty && $?.success?
+    sh "set -o pipefail && (#{command}) | xcpretty -c"
+  else
+    sh command
+  end
+end
+
+# run a command and store the output in CI artifacts
+def plain(cmd, task, subtask = '')
```

Comment:
tbh I'm not a fan of that `plain` method name.

What about `run` or `run_cmd` instead?
Another alternative (which might actually be nicer now that I think about it, at least probably worth checking how it looks) would be to wrap both the `xcpretty` and `plain` methods in a single `run_cmd` function which would take an `xcpretty: Bool` argument

Ruby 2 accepts named arguments, so we could take advantage of this to write
```
def run_cmd(cmd, task, xcpretty: false)
…
end
```
and call it using something like `run_cmd('the command to run', task, xcpretty: true)` when we want to filter using xcpretty